internal/accounts_test.go

```diff
@@ -0,0 +1,177 @@
+package internal
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/algorandfoundation/hack-tui/api"
+	"github.com/oapi-codegen/oapi-codegen/v2/pkg/securityprovider"
+	"github.com/stretchr/testify/assert"
+)
+
+func getAddressesFromGenesis(t *testing.T) ([]string, string, string) {
+
+	// TODO: replace with calls to GetGenesis
+
+	resp, err := http.Get("http://localhost:8080/genesis")
```

Comment:
>Unfortunately I had some issues with the GetGenesis API, it kept giving me a JSON unmarshalling error.

I would still grab the request using `GetGenesis` and avoid `GetGenesisWithResponses` which constructs the objects. We can look into the unmarshalling bug in the upstream definition